### PR TITLE
Make QualityControl dependent on O2

### PR DESCRIPTION
**Pull request by knopers8**
Needed for https://github.com/AliceO2Group/QualityControl/pull/7

### CODE REVIEW

File: qualitycontrol.sh [hunk 1]
```diff
@@ -31,8 +34,10 @@ cmake $SOURCEDIR                                              \
       ${MONITORING_VERSION:+-DMonitoring_ROOT=$MONITORING_ROOT} \
       ${CONFIGURATION_VERSION:+-DConfiguration_ROOT=$CONFIGURATION_ROOT} \
       ${INFOLOGGER_VERSION:+-DInfoLogger_ROOT=$INFOLOGGER_ROOT} \
+      ${O2_VERSION:+-DO2_ROOT=$O2_ROOT}                       \
```
Comment (ktf) [inline qualitycontrol.sh]: Why making it optional here and not below?

Comment (Barthelemy) [reply qualitycontrol.sh]: Actually most of them should be optional below as well (I guess) or make them non optional here.

Comment (ktf) [reply qualitycontrol.sh]: yes, agreed. @knopers8 could you fix that?

Comment (knopers8) [reply qualitycontrol.sh]: @ktf Sure. We agreed with @Barthelemy to get rid of all options here. Just give me a while to make sure it doesn't break anything

Comment (knopers8) [reply qualitycontrol.sh]: @ktf done

File: qualitycontrol.sh [hunk 2]
```diff
@@ -20,19 +21,23 @@ incremental_recipe: |
 #!/bin/bash -ex
 
 case $ARCHITECTURE in
-    osx*) [[ ! $BOOST_ROOT ]] && BOOST_ROOT=$(brew --prefix boost);;
+  osx*)
+    [[ ! $BOOST_ROOT ]] && BOOST_ROOT=$(brew --prefix boost)
+    [[ ! $GSL_ROOT ]] && GSL_ROOT=`brew --prefix gsl` ;;
```
Comment (ktf) [inline qualitycontrol.sh]: `GSL_ROOT` is not `MS_GSL_ROOT`. They are two different libraries (blame Microsoft for the aliasing...).. Is this what you really want????

Comment (knopers8) [reply qualitycontrol.sh]: Sorry, I did not know that, I copy-pasted it without enough understanding. Fixed.

**Comment by ktf** [comment]
As a general note you should try to stick to the minimum amount of changes possible when opening the PR and stick to the principle "one PR, one feature / fix". Anyways, now merged.

**Pull request merged**